pytorch_lightning/profiler/pytorch.py

```diff
@@ -102,202 +140,207 @@ def __init__(
+            sort_by_key: Attribute used to sort entries. By default
+                they are printed in the same order as they were registered.
+                Valid keys include: ``cpu_time``, ``cuda_time``, ``cpu_time_total``,
+                ``cuda_time_total``, ``cpu_memory_usage``, ``cuda_memory_usage``,
+                ``self_cpu_memory_usage``, ``self_cuda_memory_usage``, ``count``.
 
-            profiled_functions: list of profiled functions which will create a context manager on.
+            record_functions: list of profiled functions which will create a context manager on.
                 Any other will be pass through.
 
             local_rank: When running in distributed setting, local_rank is used for each process
                 to write to their own file if `output_fname` is provided.
 
+            profiler_kwargs: Keyword arguments for the PyTorch profiler. This depends on your PyTorch version
+
+            record_module_names: Whether to add module names while recording autograd operation.
+
         Raises:
             MisconfigurationException:
                 If arg ``sort_by_key`` is not present in ``AVAILABLE_SORT_KEYS``, or
                 if log file is not a ``.txt`` file.
             ValueError:
                 If you attempt to stop recording an action which was never started.
         """
+        if output_filename is not None and not output_filename.endswith(".txt"):
+            raise MisconfigurationException("`output_filename` should be a `.txt` file.")
 
-        self.profiled_actions = {}
-        self.enabled = enabled
-        self.profiled_functions = profiled_functions or self.PROFILED_FUNCTIONS
-        self.use_cuda = use_cuda
-        self.record_shapes = record_shapes
-        self.profile_memory = profile_memory
-        self.sort_by_key = sort_by_key or ("cuda_time_total" if self.use_cuda else "cpu_time_total")
-        self.with_stack = with_stack
-        self.group_by_input_shapes = group_by_input_shapes and record_shapes
-        self.use_kineto = use_kineto
-        self.use_cpu = use_cpu
+        record_functions = self.__deprecation_check(profiled_functions, record_functions)
+
+        self.output_fname = output_filename
+        self.record_functions = record_functions + list(self.RECORD_FUNCTIONS)
+        self.sort_by_key = sort_by_key or f"{'cuda' if profiler_kwargs.get('use_cuda', False) else 'cpu'}_time_total"
+        self.group_by_input_shapes = group_by_input_shapes and profiler_kwargs.get("record_shapes", False)
         self.row_limit = row_limit
         self.emit_nvtx = emit_nvtx
         self.export_to_chrome = export_to_chrome
         self.path_to_export_trace = path_to_export_trace
+        self.record_module_names = record_module_names
+        self.lightning_module = None  # set by ProfilerConnector
+        self.register = None
+        self.profiler_kwargs = profiler_kwargs
+        self.profiler = None
+        self._parent_profiler = None
```

Comment:
similar as above, are all these attributes meant to be part of the public API? or should we mark some/all as private?

Reply:
Done !